### PR TITLE
[risk=low][no ticket] User apps config panel reacts to app changes

### PR DESCRIPTION
**Pull request by jmthibault79**
The user apps config panel was checking for apps only **once** on load, instead of receiving updates by polling and checking the store.

Tested locally by opening the app config panel after creating and deleting apps, and observing that the button changed from "start" to "open" or vice-versa as appropriate **without any user interaction**

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [x] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/apps-panel.tsx [hunk 1]
```diff
@@ -151,6 +151,7 @@ export const AppsPanel = (props: {
               />
             ) : (
               <TooltipTrigger
+                key={availableApp.appType}
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel.tsx:154]: unrelated

File: ui/src/app/components/gke-app-configuration-panel.spec.tsx [hunk 2]
```diff
@@ -51,8 +56,10 @@ const validateInitialLoadingSpinner = async () => {
 
 describe(GKEAppConfigurationPanel.name, () => {
   beforeEach(() => {
+    jest.clearAllTimers();
```
Comment (jmthibault79) [inline ui/src/app/components/gke-app-configuration-panel.spec.tsx:59]: I made these 2 changes to cope with `maybeStartPollingForUserApps()`
because the tests were interfering with each other after this change

File: ui/src/app/components/gke-app-configuration-panel.spec.tsx [hunk 3]
```diff
@@ -107,20 +114,6 @@ describe(GKEAppConfigurationPanel.name, () => {
     await validateInitialLoadingSpinner();
   });
 
-  it('should show an error if the list apps API call fails', async () => {
```
Comment (jmthibault79) [inline ui/src/app/components/gke-app-configuration-panel.spec.tsx:110]: it doesn't do this anymore.

The component never calls list-apps directly but instead uses `maybeStartPollingForUserApps()` and `userAppsStore`

File: ui/src/app/utils/user-apps-utils.tsx [hunk 4]
```diff
@@ -52,7 +52,7 @@ export const updateLastActive = (userApps: ListAppsResponse) => {
       app.dateAccessed ? new Date(app.dateAccessed).valueOf() : 0
     )
   );
-  if (userAppLastActive > getLastActiveEpochMillis() ?? 0) {
+  if (userAppLastActive > (getLastActiveEpochMillis() ?? 0)) {
```
Comment (jmthibault79) [inline ui/src/app/utils/user-apps-utils.tsx:55]: unrelated - this was incorrect before the parens

File: ui/src/app/components/gke-app-configuration-panel.spec.tsx [hunk 5]
```diff
@@ -514,4 +519,51 @@ describe(GKEAppConfigurationPanel.name, () => {
       expect(screen.queryByText(deleteUnattachedPdRegex)).toBeNull();
     });
   });
+
+  it('should update the Create panel when the app changes status', async () => {
```
Comment (jmthibault79) [inline ui/src/app/components/gke-app-configuration-panel.spec.tsx:523]: new test for this case

**Review by yonghaoy — APPROVED** [review]

File: ui/src/app/components/gke-app-configuration-panel.spec.tsx [hunk 6]
```diff
@@ -13,16 +14,22 @@ import {
...
 import defaultServerConfig from 'testing/default-server-config';
 import { expectButtonElementEnabled } from 'testing/react-test-helpers';
 import {
   AppsApiStub,
   createListAppsCromwellResponse,
+  createListAppsRStudioResponse,
 } from 'testing/stubs/apps-api-stub';
 import { DisksApiStub, stubDisk } from 'testing/stubs/disks-api-stub';
 
+import { UIAppType } from './apps-panel/utils';
```
Comment (yonghaoy) [inline ui/src/app/components/gke-app-configuration-panel.spec.tsx:32]: nit: fix relative import path?

**Review by yonghaoy — APPROVED** [review]

File: ui/src/app/components/gke-app-configuration-panel.spec.tsx (hunk 3 above)
Comment (NehaBroad) [reply ui/src/app/components/gke-app-configuration-panel.spec.tsx:110]: Should we still keep this test to prove that START button remains disabled and there will be no change as userAppsStore will have an entry with ERROR state

**Review by NehaBroad — APPROVED** [review]
Tested , working well

File: ui/src/app/components/gke-app-configuration-panel.spec.tsx (hunk 3 above)
Comment (jmthibault79) [reply ui/src/app/components/gke-app-configuration-panel.spec.tsx:110]: I'm not sure that is correct ... we can talk if you like
